test/asana/helpers/test_extract_task_fields_from_pull_request.py

```diff
@@ -696,6 +769,133 @@ def test_does_not_rely_on_github_to_return_reviews_sorted_by_submitted_at_timest
         self.assertEqual(True, task_fields["completed"])
 
 
+@patch("src.dynamodb.client.get_asana_id_from_github_node_id", return_value="0")
+class TestExtractsCustomFieldsFromPullRequest(BaseClass):
```

Comment:
Let me know if these tests are in line with what you're thinking. It involved quite a bit of mocking, so if they don't accomplish exactly what you had in mind, I'm happy to take another crack at them.

Reply:
hm yeah I think I was mostly looking for unit testing `_task_status_from_pull_request`, though I think having _some_ test that checks for this custom field also makes sense - perhaps a bit of both? Have one test that checks that `"pr_status"` custom field gets set at all, then another for unit testing the logic behind `_task_status_from_pull_request`?